From 079e1cb81b6259cb4a5e651e1c02f4c3b58a20ed Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Mon, 4 May 2015 10:54:20 +0800
Subject: [PATCH] MLK-9618-5 usb: chipidea: otg: show message device no
 response for SRP fail

If b device does not detect vbus on after TB_SRP_FAIL of data pulse,
b device should show 'Device no response' message.

Signed-off-by: Li Jun <b47624@freescale.com>
---
 drivers/usb/chipidea/otg_fsm.c   | 1 +
 drivers/usb/common/usb-otg-fsm.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/usb/chipidea/otg_fsm.c b/drivers/usb/chipidea/otg_fsm.c
index 215ea57..5ad4f50 100644
--- a/drivers/usb/chipidea/otg_fsm.c
+++ b/drivers/usb/chipidea/otg_fsm.c
@@ -389,6 +389,7 @@ static int b_se0_srp_tmout(struct ci_hdrc *ci)
 static int b_srp_fail_tmout(struct ci_hdrc *ci)
 {
 	ci->fsm.b_srp_done = 1;
+	dev_warn(ci->dev, "Device No Response\n");
 	return 1;
 }
 
diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 43416c2..4600808 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -69,6 +69,7 @@ static void otg_leave_state(struct otg_fsm *fsm, enum usb_otg_state old_state)
 	switch (old_state) {
 	case OTG_STATE_B_IDLE:
 		otg_del_timer(fsm, B_SE0_SRP);
+		otg_del_timer(fsm, B_SRP_FAIL);
 		fsm->b_se0_srp = 0;
 		fsm->adp_sns = 0;
 		fsm->adp_prb = 0;
-- 
1.8.0

